tests/backup/backup_helper.go

```diff
@@ -5996,3 +5995,238 @@ func CreateKubevirtBackupRuleForAllVMsInNamespace(ctx context.Context, namespace
+			if err != nil {
+				log.Infof(fmt.Sprintf("Error creating directory %s: %s\n", dirPath, err.Error()))
+				continue
+			}
+			fileConfig := PodDirectoryConfig{
+				BasePath:          dirPath,
+				FilesPerDirectory: directoryConfig.FilesPerDirectory,
+			}
+
+			err = CreateFilesInPodDirectory(pod, fileConfig)
+			if err != nil {
+				return fmt.Errorf("error creating files in %s: %s", dirPath, err.Error())
+			}
+
+			if directoryConfig.Levels > 1 {
+				wg.Add(1)
+				go func(dirPath string, depth int) {
+					defer wg.Done()
+					DirectoryConfig := PodDirectoryConfig{
+						BasePath:          dirPath,
+						Depth:             directoryConfig.Depth,
+						Levels:            directoryConfig.Levels - 1,
+						FilesPerDirectory: directoryConfig.FilesPerDirectory,
+					}
+					errChan <- CreateDirectoryStructureInPod(pod, DirectoryConfig)
+				}(dirPath, directoryConfig.Depth)
+			}
+		}
+	}
+	go func() {
+		wg.Wait()
+		close(errChan)
+	}()
+	for err := range errChan {
+		if err != nil {
+			return fmt.Errorf("error creating nested directories: %s", err.Error())
+		}
+	}
+
+	return nil
+}
+
+// CreateDirectoryStructureInPod creates a directory structure with multiple levels and files per directory within a specified Pod.
+func CreateDirectoryStructureInPod(pod corev1.Pod, directoryConfig PodDirectoryConfig) error {
+	for i := 0; i < directoryConfig.Depth; i++ {
+		dirPath := filepath.Join(directoryConfig.BasePath, fmt.Sprintf("level_%d_depth_%d_%s", directoryConfig.Levels, i, RandomString(4)))
+		podConfig := PodDirectoryConfig{
+			BasePath: dirPath,
+		}
+		err := CreateDirectoryInPod(pod, podConfig)
+		if err != nil {
+			log.Infof(fmt.Sprintf("Error creating directory %s: %s\n", dirPath, err.Error()))
+			continue
+		}
+		fileConfig := PodDirectoryConfig{
+			BasePath:          dirPath,
+			FilesPerDirectory: directoryConfig.FilesPerDirectory,
+		}
+
+		err = CreateFilesInPodDirectory(pod, fileConfig)
+		if err != nil {
+			return fmt.Errorf("error creating files in %s: %s", dirPath, err.Error())
+		}
+
+		if directoryConfig.Levels > 1 {
+			DirectoryConfig := PodDirectoryConfig{
+				BasePath:          dirPath,
+				Depth:             directoryConfig.Depth,
+				Levels:            directoryConfig.Levels - 1,
+				FilesPerDirectory: directoryConfig.FilesPerDirectory,
+			}
+			err := CreateDirectoryStructureInPod(pod, DirectoryConfig)
+			if err != nil {
+				return fmt.Errorf("error creating nested directories: %s", err.Error())
+			}
+		}
+	}
+	return nil
+}
+
+// CreateDirectoryInPod creates a directory within a specified Pod.
+func CreateDirectoryInPod(pod corev1.Pod, directoryConfig PodDirectoryConfig) error {
+	cmd := fmt.Sprintf("mkdir -p %s ", directoryConfig.BasePath)
+	cmdArgs := []string{fmt.Sprintf("--kubeconfig=%v", CurrentClusterConfigPath), "exec", "-it", pod.Name, "-n", pod.Namespace, "--", "/bin/sh", "-c", fmt.Sprintf("%s", cmd)}
+	_, err := kubectlExec(cmdArgs)
+	if err != nil {
+		return err
+	}
```

Comment:
Can we use core.Instance().RunCommandInPod?

Reply:
Added core.Instance().RunCommandInPod instead of kubeExec.Thanks for suggestion @kphalgun-px , its much faster.